src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Rules/AttributesMustMatch.cs

```diff
@@ -0,0 +1,357 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules
+{
+    /// <summary>
+    /// This class implements a rule to check that the attributes between public members do not change.
+    /// </summary>
+    public class AttributesMustMatch : IRule
+    {
+        private readonly RuleSettings _settings;
+        private readonly HashSet<string> _attributesToExclude = new();
+
+        public AttributesMustMatch(RuleSettings settings, IRuleRegistrationContext context, IEnumerable<string>? excludeAttributesFiles)
+        {
+            _settings = settings;
+            if (excludeAttributesFiles != null)
+            {
+                readExclusions(excludeAttributesFiles);
+            }
+            context.RegisterOnMemberSymbolAction(RunOnMemberSymbol);
+            context.RegisterOnTypeSymbolAction(RunOnTypeSymbol);
+        }
+
+        private void readExclusions(IEnumerable<string> excludeAttributesFiles)
```

Comment:
nit: we don't use lowercase names in C#

```suggestion
        private void ReadExclusions(IEnumerable<string> excludeAttributesFiles)
```